src/ipc/ipc4/handler.c

```diff
@@ -1325,7 +1326,11 @@ struct ipc_cmd_hdr *ipc_prepare_to_send(const struct ipc_msg *msg)
 void ipc_boot_complete_msg(struct ipc_cmd_hdr *header, uint32_t data)
 {
 	header->pri = SOF_IPC4_FW_READY;
-	header->ext = 0;
+
+	if (IS_ENABLED(CONFIG_ADSP_IMR_CONTEXT_SAVE))
+		header->ext = SOF_IPC_INFO_D3_PERSISTENT;
+	else
+		header->ext = 0;
```

Comment:
@mwasko @marcinszkudlinski please comment. Is this ok, or any other way to rely this information to host in IPC4?

Reply:
Not following why we are using the same flag for different firmware behavior.

Reply:
Introduce a better flag : SOF_IPC4_FW_CONTEXT_SAVE

Reply:
@RanderWang @plbossart @ujfalusi lets make sure we add in more useful flags here if needed, i.e. we should be able to determine imporant vendor FW features and config based on the FW Kconfig that wont be in manifest.